src/components/Input.tsx

```diff
@@ -0,0 +1,51 @@
+import React, {useState} from "react";
+import styled from "styled-components/macro";
+import { Theme } from "styles/theme";
+import {Box, BoxStyles, Typography, FlexWrapper} from 'components';
+
+export type InputProp = 'text' | 'number' | 'password' | 'email' | 'date';
+
+interface InputProps extends React.InputHTMLAttributes<HTMLIFrameElement> {
+    id: string;
+    minValue?: number;
+    maxValue?: number;
+    placeholder?: string;
+    type: InputProp;
+    onChange?: ()=> void;
+    border?: string;
+    borderRadius?: string;
+}
+interface InputWrapProps extends InputProps {
+    label?: string;
+    errorMessage?: string;
+
+}
+
+export const InputWrap: React.FC<InputWrapProps> = ({
+    id,
+    minValue,
+    maxValue,
+    placeholder,
+    type,
+    border, 
+    borderRadius,
+    label, 
+    errorMessage, 
+    // ...rest // "This spread always overwrites this property."
+}) => {
+    const [isError, setIserror] = useState(true);
+    const [value, setValue] = useState<string>("");
```

Comment:
How will you use this value? there should be no logic for onChange method. all you have to do - pass the prop to onChange attribute.
Input shoud accept these as a prop and show it.